### ISSUE TITLE
The gradient of the hypernetwork is very small.

### ISSUE THREAD

**Issue by UkiTenzai** [t2ance/MQVAE#2]
Hello, I downloaded your code and used CIFAR for training. Apart from modifying the Huggingface and W&B tokens and setting `"dataset": "cifar10"`, I did not make any other changes. During training, I wanted to observe the training process of the hypernetwork, so I added `wandb.watch(codebook_module, log="all", log_freq=1)` at line 364 in training_vqgan.py. However, I noticed that the gradient of the hypernetwork, called codebook_module, is very small, and its parameters are almost not updating, even though the reconstruction loss (recloss) is still decreasing.
As shown in the image below, a gradient is only available every 400 iterations.
![Image](https://github.com/user-attachments/assets/1aed01b4-fe55-40f3-b06f-206e76055ed9)
![Image](https://github.com/user-attachments/assets/5c8a8249-d49e-4dea-bf81-9c048f8789d3)

At first, I thought the issue might be with "gradient_accumulation": 4, so I changed it to 1. However, I found that the gradient is still very small, and the parameters are almost not updating, just like in the image below.
![Image](https://github.com/user-attachments/assets/c265a615-d907-46aa-bd0e-a73393ec4db5)
![Image](https://github.com/user-attachments/assets/0655b7b1-e139-4fa8-ba0d-902cfccd4fc4)

Could you please advise on how to fix this bug?

Additionally, regardless of the bug here, I believe this work is very valuable and has been very inspiring to me. Congratulations on the acceptance of this work to ICML.

**Comment by t2ance** [comment]
Hi, thanks for your interest! Could you let me know about your hyperparameter setting, or share with me your wandb run so that I can better help you?

**Comment by UkiTenzai** [comment]
Sure here is the source code and wandb link (ignore the token inside which are read only
Those figures are from runs/r1ryshkr

[MQVAE-master.zip](https://github.com/user-attachments/files/20968012/MQVAE-master.zip)

https://wandb.ai/parslog/vqgan?nw=nwuserparslog

**Comment by t2ance** [comment]
Hi, I think the scale of gradients is okay. Basically, because the mechanism is to generate a codebook from a forward pass of a hyper-network, and the hyper-network is not directly used for quantization. It should be okay for it to only change on a small scale, but result in a very different codebook.

By the way, for the CIFAR10 dataset, we used the VQVAE architecture in our paper. VQGAN architecture seems too large for this dataset, and you can set it by setting the generator_type to "vqvae".

**Comment by UkiTenzai** [comment]
I see. Thanks
